patches/api/0416-Add-whitelist-events.patch

```diff
@@ -0,0 +1,124 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: SageSphinx63920 <sage@sagesphinx63920.dev>
+Date: Sun, 14 May 2023 12:56:15 +0200
+Subject: [PATCH] Add whitelist events
+
+
+diff --git a/src/main/java/io/papermc/paper/event/player/PlayerWhitelistAddEvent.java b/src/main/java/io/papermc/paper/event/player/PlayerWhitelistAddEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..7d788bdd52d63f96c56f0a13cddd6101d29e8dc7
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/PlayerWhitelistAddEvent.java
+@@ -0,0 +1,53 @@
++package io.papermc.paper.event.player;
++
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.Event;
++import org.bukkit.event.HandlerList;
++import org.jetbrains.annotations.NotNull;
++
++import java.util.UUID;
++
++/**
++ * Called when a player is being added to the whitelist
++ */
++public class PlayerWhitelistAddEvent extends Event implements Cancellable {
++
++    private static final HandlerList HANDLER_LIST = new HandlerList();
++    private boolean cancel = false;
++    @NotNull private final UUID who;
++
++    public PlayerWhitelistAddEvent(@NotNull UUID who) {
++        this.who = who;
++    }
++
++    /**
++     * Gets the UUID of the player being added to the whitelist
++     *
++     * @return the UUID of the player being added to the whitelist
++     */
++    @NotNull
++    public UUID getUUID() {
++        return who;
++    }
++
++    @Override
++    public boolean isCancelled() {
++        return cancel;
++    }
++
++    @Override
++    public void setCancelled(boolean cancel) {
++        this.cancel = cancel;
++    }
++
++    @NotNull
++    @Override
++    public HandlerList getHandlers() {
++        return HANDLER_LIST;
++    }
++
++    @NotNull
++    public static HandlerList getHandlerList() {
++        return HANDLER_LIST;
++    }
++}
+diff --git a/src/main/java/io/papermc/paper/event/player/PlayerWhitelistRemoveEvent.java b/src/main/java/io/papermc/paper/event/player/PlayerWhitelistRemoveEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..1c8c392646498ab09e0721e1d00ba8e3c7c47992
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/PlayerWhitelistRemoveEvent.java
+@@ -0,0 +1,53 @@
++package io.papermc.paper.event.player;
```

Comment:
I'd argue this package is not really the best package for the event.
The player package usually holds events caused by in-game players.

This does not reference such a player but rather just their resolved uuid.
The event.server package might be better ?

Reply:
I talked about this before creating the event in this package. I could also call Bukkit.getPlayer which would give me the player. The event is only triggered for a player but calling the method would add some extra server load

Reply:
I think getting the Player should be up to the plugins really.

Reply:
Yeah that's my thought